### DISCUSSION TITLE
How to extend session cookie?

### DISCUSSION THREAD

**Discussion by N-point-N** [zitadel/zitadel#8897] [Q&A]
I'm using zitadel golang library for web app authorization, it works pretty well, but users have to re-login manually several times per day
I've found this place, that can be root cause 
https://github.com/zitadel/zitadel-go/blob/main/pkg/authentication/authenticate.go#L205

 Here cookie age is always set as 0, so I miss session id in cookie every time
 Could you please help with some solution for avoiding this problem?

**Comment by fforootd** [comment]
Did you connect your app to an IdP like Zitadel? Or is this app a standalone?

Reason I am asking is that session lifecycle could also be done on the IdP side, which Zitadel does.
Or on other words if your apps session expires it will re-authenticate with the IdP and if that has an open session a user can automatically resume its work.
